### ISSUE TITLE
New Installation, old user

### ISSUE THREAD

**Issue by dlraymer** [apprenticeharper/DeDRM_tools#242]
I recently had to reinstall Windows 10.  I have reinstalled Calibre, and all my books are in the proper libraries.  I installed 6.5.3.  When I click on Adobe Digital Editions, the green arrow, I get an error message that the default encryption key cannot be found.  I apologize, for years I have just reinstalled the tool and it worked.  Can you tell me how to get my default encryption key?

Thanks,

Debra

**Comment by ElleKayEm** [comment]
Do you have Adobe Digital Editions authorized with your Adobe ID?

**Comment by dlraymer** [comment]
Yes.  Here is the debug info

calibre 2.82 [64bit]  embedded-python: True is64bit: True
Windows-8-6.2.9200 Windows ('64bit', 'WindowsPE')
('Windows', '8', '6.2.9200')
Python 2.7.9
Windows: ('8', '6.2.9200', '', 'Multiprocessor Free')
Successfully initialized third party plugins: DeDRM (6, 5, 3) && Kindle Collections (1, 7, 17) && Reading List (1, 6, 6)
USB devices on system:
(USBDevice(vendor_id=0x2687 product_id=0xfb01 bcd=0x100 devid=usb\vid_2687&pid_fb01&rev_0100 devinst=2),
 USBDevice(vendor_id=0x4f9 product_id=0x270 bcd=0x100 devid=usb\vid_04f9&pid_0270&rev_0100&mi_01 devinst=3),
 USBDevice(vendor_id=0x4f9 product_id=0x270 bcd=0x100 devid=usb\vid_04f9&pid_0270&rev_0100 devinst=4),
 USBDevice(vendor_id=0xbc2 product_id=0xab2d bcd=0x100 devid=usb\vid_0bc2&pid_ab2d&rev_0100 devinst=5),
 USBDevice(vendor_id=None product_id=None bcd=None devid=usb\root_hub30&vid8086&pida12f&rev0031 devinst=6),
 USBDevice(vendor_id=0x4f2 product_id=0x1459 bcd=0x100 devid=usb\vid_04f2&pid_1459&rev_0100 devinst=7),
 USBDevice(vendor_id=0x461 product_id=0x4e67 bcd=0x112 devid=usb\vid_0461&pid_4e67&rev_0112 devinst=8),
 USBDevice(vendor_id=0x461 product_id=0x4e67 bcd=0x112 devid=usb\vid_0461&pid_4e67&rev_0112&mi_01 devinst=9),
 USBDevice(vendor_id=0x461 product_id=0x4e67 bcd=0x112 devid=usb\vid_0461&pid_4e67&rev_0112&mi_00 devinst=10),
 USBDevice(vendor_id=0x4f9 product_id=0x270 bcd=0x100 devid=usb\vid_04f9&pid_0270&rev_0100&mi_02 devinst=11),
 USBDevice(vendor_id=0xbda product_id=0x153 bcd=0x5713 devid=usb\vid_0bda&pid_0153&rev_5713 devinst=12),
 USBDevice(vendor_id=0x2687 product_id=0xfb01 bcd=0x100 devid=usb\vid_2687&pid_fb01&rev_0100&mi_00 devinst=13),
 USBDevice(vendor_id=0x4f9 product_id=0x270 bcd=0x100 devid=usb\vid_04f9&pid_0270&rev_0100&mi_00 devinst=14),
 USBDevice(vendor_id=0x2687 product_id=0xfb01 bcd=0x100 devid=usb\vid_2687&pid_fb01&rev_0100&mi_01 devinst=15))

No disabled plugins
Looking for devices of type: MTP_DEVICE
List of WPD PNP ids:
[u'\\\\?\\swd#wpdbusenum#{bdaa3427-0b5a-11e7-8049-dcfe070513d0}#0000000000100000#{6ac27878-a6fa-4155-ba85-f98f491d4f33}']
No suitable MTP devices found
 
Looking for devices of type: SMART_DEVICE_APP
All IP addresses {'{5139254B-0B6E-11E7-8045-806E6F6E6963}': [{'broadcast': '127.255.255.255', 'netmask': '255.0.0.0', 'addr': '127.0.0.1'}], '{A06EF5A4-86BE-45BC-B9B6-289DA0455EBE}': [{'broadcast': '192.168.1.255', 'netmask': '255.255.255.0', 'addr': '192.168.1.95'}], '{54835A8B-EA8E-47F1-B3DB-FC07E72B7D8F}': [{'addr': '169.254.76.244'}], '{64492727-4F84-49B2-9C72-8A188DBD1C07}': [{'addr': '169.254.0.251'}]}
No device is connected
 
Looking for devices...
 
Devices possibly connected: None

**Comment by ElleKayEm** [comment]
Did you mean to close this issue?

If ADE is on the same machine and user as calibre, then importing an Adobe DRM epub should retrieve your key automatically.  Also, it's best to use ADE 2.0.1, or you might get DRM that can't be removed.

**Comment by dlraymer** [comment]
No, I didn't mean to close it.  ADE is on the same machine.  I import it, but it says it has DRM.  Never got this message before.  I looked, and it said to look at the green plus sign, but that says no key found.

Any help would be very much appreciated.

**Comment by dlraymer** [comment]
I copied my old digital editions file onto a backup drive.  I had to reinstall ade and calibre.  I just tried to import a file that had successfully imported into my old calibre without drm.  It did not.

I like to buy epub books, because it seems to be a universal format, but if I can no longer remove drm and convert them, I need to change my buying habits.  I would appreciate any help

**Comment by dlraymer** [comment]
I am using 2.0.1.7865

**Comment by dlraymer** [comment]
I bought a new book and downloaded it, and it still had drm.   I downloaded a kindle book and imported that, and it did not have drm.  I have several ade  data files on my computer.  I am going to delete all of them and delete ade and reinstall it.

**Comment by dlraymer** [comment]
I deleted all my ade data files, uninstalled ade and removed the drm plug in from Calibre.  Then I reinstalled ade, downloaded a book, and installed drm 6.5.2 which is what I had on my machine before I had to reinstall windows.  Still no luck.

Aside from email, Calibre is the most Important thing I do with my computer.  If I can't get this to work, I think I will reinstall windows again.  If someone could tell me the order to do everything, so this doesn't happen again, I would really appreciate it.

**Comment by dlraymer** [comment]
Here is a better debug file.  Seems to be a problem with pyro?


[tmp1hswmy.txt](https://github.com/apprenticeharper/DeDRM_tools/files/853798/tmp1hswmy.txt)

**Comment by ElleKayEm** [comment]
No, the problem is definitely that it isn't able to retrieve the key for some reason.  Let me see if I can get someone else to take a look at this.

**Comment by dlraymer** [comment]
I tried to figure out what was different, as I only got this computer a year ago, and this installed fine at that time.  So I uninstalled ade and reinstalled with Chrome.  Then I went ahead and downloaded pycripto and python.  Still no luck.  
[#3.txt](https://github.com/apprenticeharper/DeDRM_tools/files/855790/3.txt)

**Comment by dlraymer** [comment]
I downloaded ADE 2.0 from the adobe site.  Is it possible they put something in it preventing discovery of the key?  Is there someplace else I should download it from?

**Comment by dlraymer** [comment]
Would my Adobe ID allow me to get the key?

**Comment by dlraymer** [comment]
When I run ineptepub.py, this is what I get.  Does that help?

![image](https://cloud.githubusercontent.com/assets/26526266/24127659/771ab71c-0da4-11e7-9a30-de200ddb4809.png)

**Comment by dlraymer** [comment]
SUCCESS!!!  I uninstalled ADE, removed the registry key, restarted the machine and reinstalled ADE.  I had to change the password because I had forgotten my old one, and for some reason that must have screwed something up in the registry.  I still have one book that wont download into ADE, but I already have that book in Calibre and converted.

THANK YOU, ELLEKAYEM!!  I am retired, and I like to do 4 things-knit, cook, work out, and hack books.
